crates/runtime/src/dataconnector/github.rs

```diff
@@ -0,0 +1,324 @@
+                        }}
+                    }}
+                }}
+            }}"#,
+            owner = self.owner,
+            name = self.repo
+        );
+
+        (query, "/data/repository/issues/nodes".to_string(), 1)
+    }
+}
+
+pub fn args_from_ds(
+    ds: &Dataset,
+) -> Result<Arc<dyn GithubTableArgs>, Box<dyn Error + Send + Sync>> {
+    let path = ds.path().clone();
+    let mut parts = path.split('/');
+
+    match (parts.next(), parts.next(), parts.next(), parts.next()) {
+        (Some("github.com"), Some(owner), Some(repo), Some("pulls")) => {
+            Ok(Arc::new(PullRequestTableArgs {
+                owner: owner.to_string(),
+                repo: repo.to_string(),
+            }))
+        }
+        (Some("github.com"), Some(owner), Some(repo), Some("commits")) => {
+            Ok(Arc::new(CommitTableArgs {
+                owner: owner.to_string(),
+                repo: repo.to_string(),
+            }))
+        }
+        (Some("github.com"), Some(owner), Some(repo), Some("issues")) => {
+            Ok(Arc::new(IssueTableArgs {
+                owner: owner.to_string(),
+                repo: repo.to_string(),
+            }))
+        }
+        (Some("github.com"), Some(_), Some(_), Some(invalid_table)) => {
+            Err(format!("Invalid Github table type: {invalid_table}").into())
+        }
+        (_, Some(owner), Some(repo), _) => {
+            Err(format!("`from` must start with 'github.com/{owner}/{repo}'").into())
+        }
+        _ => Err("Invalid Github dataset path".into()),
+    }
+}
+
+impl Github {
+    pub(crate) fn create_graphql_client(
+        &self,
+        tbl: &Arc<dyn GithubTableArgs>,
+    ) -> std::result::Result<GraphQLClient, Box<dyn std::error::Error + Send + Sync>> {
+        let access_token = self.params.get("access_token").expose().ok();
+
+        let Some(endpoint) = self.params.get("endpoint").expose().ok() else {
+            return Err("Github 'access_token' not provided".into());
+        };
+
+        let client = default_spice_client("application/json").boxed()?;
+
+        let (query, json_pointer, unnest_depth) = tbl.get_graphql_values();
+
+        Ok(GraphQLClient::new(
+            client,
+            Url::parse(endpoint).boxed()?,
+            query,
+            json_pointer,
+            access_token,
+            None,
+            None,
+            unnest_depth,
+        ))
+    }
+}
+
+#[derive(Default, Copy, Clone)]
+pub struct GithubFactory {}
+
+impl GithubFactory {
+    #[must_use]
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    #[must_use]
+    pub fn new_arc() -> Arc<dyn DataConnectorFactory> {
+        Arc::new(Self {}) as Arc<dyn DataConnectorFactory>
+    }
+}
+
+const PARAMETERS: &[ParameterSpec] = &[
+    ParameterSpec::connector("access_token")
+        .description("A Github access token.")
```

Comment:
You can mark these as required here, and the common logic to check if they are present will kick in

Reply:
Technically not required. Will just operate at a lower rate limit. Should we require as mandatory for better DX?

Reply:
Ah, I thought I saw somewhere that we checked for it. If its not required then we don't need to mark it as such here.